File: weekend-core/src/main/java/com/github/aly8246/core/annotation/WeekendDaoScan.java
package com.github.aly8246.core.annotation;

import com.github.aly8246.core.proxy.WeekendProxyRegister;
import org.springframework.context.annotation.Import;

import java.lang.annotation.*;

@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.TYPE)
@Documented
@Import(WeekendProxyRegister.class)
public @interface WeekendDaoScan {
String[] value() default {};
}

File: weekend-core/src/main/java/com/github/aly8246/core/annotation/WeekendDaoScan.kt
package com.github.aly8246.core.annotation

import com.github.aly8246.core.proxy.WeekendProxyRegister
import org.springframework.context.annotation.Import

@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.CLASS, AnnotationTarget.FILE)
@MustBeDocumented
@Import(WeekendProxyRegister::class)
annotation class WeekendDaoScan(vararg val value: String = [])
